src/lib/pipeline.ml

```diff
@@ -214,33 +225,49 @@ module Full (Bfx: Extended_edsl.Semantics) = struct
       List.map vcfs ~f:(fun (name, s, v) -> name, s, Bfx.filter_to_region v bed)
 
 
-  let qc fqs =
-    Bfx.concat fqs |> Bfx.fastqc
-
-  let rna_bam ~parameters ~reference_build fqs =
-    Bfx.list_map fqs
-      ~f:(Bfx.lambda (fun fq ->
-          Bfx.star ~configuration:star_config ~reference_build fq))
+  let qc fqs = Bfx.concat fqs |> Bfx.fastqc
+
+  (* Makes a list of samples (which are themselves fastqs or BAMs) into one (or
+     two, if paired-end) FASTQs. *)
+  let concat_samples samples =
+    Bfx.list_map
+      ~f:(Bfx.lambda (fun f -> Bfx.concat f))
+      (Bfx.list samples)
+
+  let rna_bam ~parameters ~reference_build samples =
+    let sample_to_bam sample =
+      Bfx.list_map sample
+        ~f:(Bfx.lambda (fun fq ->
+            Bfx.star ~configuration:star_config ~reference_build fq))
+      |> Bfx.merge_bams
+      |> Bfx.picard_mark_duplicates
+        ~configuration:(mark_dups_config parameters.Parameters.picard_java_max_heap)
+      |> Bfx.gatk_indel_realigner
```

Comment:
"So why do we do #3 [independent realignment] in production? Because the gains from the other methods are not very large, and introduce more complexity and compute requirements than they're worth. As long as you use HaplotypeCaller (and its awesome reassembly functionality) for the variant calling, you don't really need the best possible indel realignment -- you just need it to be good enough to iron out most of the artifacts that would otherwise add noise to the BQSR model."

I don't think MuTect (1) or Strelka do local reassembly, though?